cpp/src/lists/set_operations.cu

```diff
@@ -163,13 +163,13 @@ std::unique_ptr<column> intersect_distinct(lists_column_view const& lhs,
     rmm::mr::get_current_device_resource());
 
   // A stable algorithm is required to ensure that list labels remain contiguous.
-  auto out_table = cudf::detail::stable_distinct(intersect_table->view(),
-                                                 {0, 1},  // indices of key columns
-                                                 duplicate_keep_option::KEEP_ANY,
-                                                 nulls_equal,
-                                                 nans_equal,
-                                                 stream,
-                                                 mr);
+  auto out_table = cudf::stable_distinct(intersect_table->view(),
```

Comment:
I believe we are supposed to use the detail API and not the public API here. We typically call detail APIs internally to avoid triggering additional logic that is only meant to wrap the public API.

Reply:
I presumed we wanted to use public where possible so we would get nvtx support for calling algorithms

Reply:
Dropped all these instances after offline discussion